lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -67,25 +67,32 @@ defmodule Thrift.Generator.StructBinaryProtocol do
   At the moment it also generates an experimental serializer that may be faster.
   """
   def struct_deserializer(%{fields: fields}, name, file_group) do
-    field_matchers = Enum.map(fields, fn %Field{name: name} ->
-      {name, Macro.var(name, nil)}
-    end)
-
-    struct_matcher = {:%, [], [name, {:%{}, [], field_matchers}]}
-
-    field_serializers = Enum.map(fields, fn %Field{name: name, type: type, id: id} ->
-      var = Macro.var(name, nil)
-      quote do
-        case unquote(var) do
-          nil ->
-            <<>>
-          _ ->
-            unquote([
-              quote do <<unquote(type_id(type, file_group)), unquote(id) :: size(16)>> end,
-              value_serializer(type, var, file_group)
-            ] |> Utils.merge_binaries |> Utils.simplify_iolist)
+    fields = Enum.reject(fields, &(&1.type == :void))
+
+    struct_matcher = case fields do
+                       [] ->
```

Comment:
My comments are more a function of me reading through the diffs from top to bottom for each iteration. It's a large amount of new code, and inconsistency and extra whitespace tend to pop out during those readings.